ifElse/main.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import "fmt"
+
+func main() {
+	// In Go, no parentheses around conditions required
+	// but the braces.
+	if 7%2 == 0 {
+		fmt.Println("7 is even")
+	} else {
+		fmt.Println("7 is odd")
+	}
+
+	if 8%4 == 0 {
+		fmt.Println("8 is devisible by 4 !!!")
+	}
+
+	// A statement can precede conditionals;
+	// any variables declared in the statement are available in all branches. 
+	if num := 9; num < 0 {
+		fmt.Println(num, "is negative")
+	} else if num < 10 {
+		fmt.Println(num, "has 1 digit")
+	} else {
+		fmt.Println(num, "has multiple digits")
+	}
+
+	// Above can be written as this.
+	num := 7
```

Comment:
The two formats, `if foo := "something"; someBoolExpression {}` and
```go
foo := "something";
if someBoolExpression {}
```
are *not* equivalent.

In the first one `foo` is only available inside the `if`s or `else`s, where as the last one `foo` is available after the conditionals as well.